components/backup-stream/tests/mod.rs

```diff
@@ -628,26 +628,16 @@ mod test {
                 .global_progress_of_task("test_fatal_error"),
         )
         .unwrap();
-        assert_eq!(safepoints.len(), 4, "{:?}", safepoints);
+
         assert!(
-            safepoints
-                .iter()
-                .take(3)
-                // They are choosing the lock safepoint, it must greater than the global checkpoint.
-                .all(|sp| { sp.safepoint.into_inner() >= checkpoint }),
+            safepoints.iter().any(|sp| {
+                sp.serivce.contains(&format!("{}", victim))
```

Comment:
maybe use `partition_in_place` by `sp.serivce.contains(&format!("{}", victim)` and check tow parts is more precise.

Reply:
The problem is the flush procedure is asynchronous: which means it sometime won't take place. Checking them may get failed and is out of the scope of `fatal_error`.